Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift

```diff
@@ -24,149 +23,224 @@ struct LineLengthRule: Rule {
+        private var commentOnlyLines = Set<Int>()
+        private var interpolatedStringLines = Set<Int>()
+        private var multilineStringLines = Set<Int>()
 
-            if configuration.ignoresFunctionDeclarations &&
-                lineHasKinds(line: line,
-                             kinds: functionKinds,
-                             kindsByLine: swiftDeclarationKindsByLine.value) {
-                return nil
+        override func visit(_ node: SourceFileSyntax) -> SyntaxVisitorContinueKind {
+            // Populate functionDeclarationLines if ignores_function_declarations is true
+            if configuration.ignoresFunctionDeclarations {
+                let funcVisitor = FunctionLineVisitor(locationConverter: locationConverter)
+                funcVisitor.walk(node)
+                functionDeclarationLines = funcVisitor.lines
             }
 
-            if configuration.ignoresComments &&
-                lineHasKinds(line: line,
-                             kinds: commentKinds,
-                             kindsByLine: syntaxKindsByLine.value) &&
-                !lineHasKinds(line: line,
-                              kinds: nonCommentKinds,
-                              kindsByLine: syntaxKindsByLine.value) {
-                return nil
+            // Populate multilineStringLines if ignores_multiline_strings is true
+            if configuration.ignoresMultilineStrings {
+                let stringVisitor = MultilineStringLiteralVisitor(locationConverter: locationConverter)
+                stringVisitor.walk(node)
+                multilineStringLines = stringVisitor.linesSpanned
             }
 
-            if configuration.ignoresInterpolatedStrings &&
-                lineHasKinds(line: line,
-                             kinds: [.stringInterpolationAnchor],
-                             kindsByLine: syntaxKindsByLine.value) {
-                return nil
+            // Populate interpolatedStringLines if ignores_interpolated_strings is true
+            if configuration.ignoresInterpolatedStrings {
+                let interpVisitor = InterpolatedStringLineVisitor(locationConverter: locationConverter)
+                interpVisitor.walk(node)
+                interpolatedStringLines = interpVisitor.lines
             }
 
-            if configuration.ignoresMultilineStrings &&
-               lineIsMultilineString(line, file: file, syntaxKindsByLine: syntaxKindsByLine.value) {
-                return nil
+            // Populate commentOnlyLines if ignores_comments is true
+            if configuration.ignoresComments {
+                commentOnlyLines = findCommentOnlyLines(in: node, file: file, locationConverter: locationConverter)
             }
 
-            for pattern in configuration.excludedLinesPatterns where line.containsMatchingPattern(pattern) {
-                return nil
+            return .skipChildren // We'll do the main processing in visitPost
```

Comment:
Processing could be done right here as well. But the separation of information collection and processing is actually quite nice. Doing everything in one function would spare all the `*Lines` properties, though.

Reply:
Going to keep this as-is.